bundles/core/org.eclipse.smarthome.core/src/main/java/org/eclipse/smarthome/core/net/NetUtil.java

```diff
@@ -187,6 +190,50 @@ public static String getBroadcastAddress() {
     }
 
     /**
+     * Gets every IPv4+IPv6 Address on each Interface except the loopback
+     * If addPrefix is true, the Address format is in the CIDR notation
+     * which is ip/prefix-length e.g. 129.31.31.1/24 otherwise only the IP is returned.
+     *
+     * @param addPrefix Add the prefix to each returned IP address
+     * @return The collected IPv4 and IPv6 Addresses
+     */
+    public static Set<String> getInterfaceAddresses(boolean addPrefix) {
+        Set<String> interfaceIPs = new HashSet<>();
+
+        Enumeration<NetworkInterface> en;
+        try {
+            en = NetworkInterface.getNetworkInterfaces();
+        } catch (SocketException ex) {
+            LOGGER.error("Could not find interface IP addresses: {}", ex.getMessage(), ex);
+            return interfaceIPs;
+        }
+
+        while (en.hasMoreElements()) {
+            NetworkInterface networkInterface = en.nextElement();
+            boolean isLoopback = true;
```

Comment:
It does not care, I just wonder if you need "isLoopback" on your initial implementation...

```Java
boolean isLoopback = true;
try {
    isLoopback = networkInterface.isLoopback();
} catch (SocketException ignored) {
}
if (isLoopback) {
    continue;
}
```

Why not shrink it to this one:

```Java
try {
    if (networkInterface.isLoopback()) {
        continue;
    }
} catch (SocketException ignored) {
}
```

Reply:
`NetworkConfigOptionProvider` in core.net.internal does it like this:

```
try {
            final Enumeration<NetworkInterface> interfaces = NetworkInterface.getNetworkInterfaces();
            while (interfaces.hasMoreElements()) {
                final NetworkInterface current = interfaces.nextElement();
                if (!current.isUp() || current.isLoopback() || current.isVirtual()) {
                    continue;
                }

                for (InterfaceAddress ifAddr : current.getInterfaceAddresses()) {
                    InetAddress addr = ifAddr.getAddress();

                    if (addr.isLoopbackAddress() || (addr instanceof Inet6Address)) {
                        continue;
                    }

                    @SuppressWarnings("null")
                    @NonNull
                    String ipv4Address = addr.getHostAddress();
                    try {
                        String subNetString = NetUtil.getIpv4NetAddress(ipv4Address, ifAddr.getNetworkPrefixLength())
                                + "/" + String.valueOf(ifAddr.getNetworkPrefixLength());
                        subnets.add(subNetString);
                    } catch (IllegalArgumentException ex) {
                        logger.error("Could not calculate network address: {} Ignoring IP {}", ex.getMessage(),
                                ipv4Address, ex);
                    }
                }
            }
        } catch (SocketException ex) {
            logger.error("Could not retrieve network interface: {}", ex.getMessage(), ex);
            return null;
        }
```

They even check for "up" and "not virtual"

Reply:
You are right. Is there is one SocketException the whole loop is stopped. Locally exception handling would be better.